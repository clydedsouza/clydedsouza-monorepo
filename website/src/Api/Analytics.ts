import mixpanel from 'mixpanel-browser'
import { PageTypes } from '../Components/Navigation/PageTypes'
import {
  ILinkClickedProps,
  MIXPANEL_DEV_PROJECT_ID,
  MIXPANEL_PROD_PROJECT_ID,
} from './IAnalytics'

export const initAnalyticsWithSuperProperties = () => {
  const mixpanelProjectId =
    process.env.NODE_ENV === 'production'
      ? MIXPANEL_PROD_PROJECT_ID
      : MIXPANEL_DEV_PROJECT_ID

  try {
    mixpanel.init(mixpanelProjectId)
    mixpanel.register({
      environment: process.env.NODE_ENV,
    })
  } catch {
    undefined
  }
}

export const sendPageViewEvent = (pageType: PageTypes) => {
  try {
    mixpanel.track('Page view', { page: pageType })
  } catch {
    undefined
  }
}

export const sendLinkClickedEvent = (props: ILinkClickedProps) => {
  try {
    mixpanel.track('Link clicked', {
      destination_url: props.link,
      link_type: props.type,
      page_location: props.location,
    })
  } catch {
    undefined
  }
}
