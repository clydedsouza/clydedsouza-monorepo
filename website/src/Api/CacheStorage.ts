import { PageTypes } from '../Components/Navigation/PageTypes'
import { IProjectData } from './IProjectData'

export const projectDataCacheMap: Map<PageTypes, IProjectData> = new Map()
