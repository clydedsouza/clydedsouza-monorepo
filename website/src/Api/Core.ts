import axios from 'axios'
import { PageTypes } from '../Types/PageTypes'
import { IProjectData } from '../Types/ProjectData'

const BASE_URL = 'https://api.clydedsouza.net/'

const getProjectApiUrl = (pageType: PageTypes) => {
  switch (pageType) {
    case PageTypes.Books:
      return `${BASE_URL}books.json`
    case PageTypes.Speaking:
      return `${BASE_URL}speaking.json`
    case PageTypes.Teaching:
      return `${BASE_URL}teaching.json`
    case PageTypes.Projects:
    default:
      return `${BASE_URL}pinned.json`
  }
}

export const getProjectData = (pageType: PageTypes): Promise<IProjectData> => {
  const request = getProjectApiUrl(pageType)
  return axios
    .get(request)
    .then((response) => {
      return response.data
    })
    .catch(() => {
      throw new Error('API error')
    })
}
