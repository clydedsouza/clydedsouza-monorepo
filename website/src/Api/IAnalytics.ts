export const MIXPANEL_DEV_PROJECT_ID = '6c905af7861c30571d76bbc6d67696ef'
export const MIXPANEL_PROD_PROJECT_ID = '50f70ae3a52dd646aea4025dc1f30a06'

export enum AnalyticsLinkType {
  Cta = 'Cta',
  SocialIcons = 'Social icons',
  CardGitHub = 'GitHub',
  CardWebsite = 'Website',
  GeneralWebsite = 'External website',
}

export interface ILinkClickedProps {
  link?: string
  type?: AnalyticsLinkType
  location?: string
}
