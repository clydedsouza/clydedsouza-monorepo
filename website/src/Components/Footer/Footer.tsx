import Cta from '../Cta/Cta'
import './Footer.scss'

function Footer() {
  return (
    <>
      <footer>
        <Cta />
        <p>Website crafted by Clyde DSouza</p>
      </footer>
    </>
  )
}

export default Footer
