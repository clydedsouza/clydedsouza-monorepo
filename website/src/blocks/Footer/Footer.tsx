import Cta from "@/blocks/Cta/Cta";
import "./Footer.scss";
import { GoogleAds } from "../GoogleAds/GoogleAds";

function Footer() {
  return (
    <>
      <footer>
        <GoogleAds />
        <p>Website crafted by Clyde D&#39;Souza</p>
      </footer>
    </>
  );
}

export default Footer;
