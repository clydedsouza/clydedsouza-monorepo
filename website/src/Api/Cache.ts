import { PageTypes } from '../Types/PageTypes'
import { IProjectData } from '../Types/ProjectData'
import { getProjectData } from './Core'

const projectDataCacheMap: Map<PageTypes, Promise<IProjectData>> = new Map()

export const getCachedProjectData = (
  pageType: PageTypes
): Promise<IProjectData> => {
  if (!projectDataCacheMap.has(pageType)) {
    const getProjectDataPromise = getProjectData(pageType).catch((error) => {
      projectDataCacheMap.delete(pageType)
      return Promise.reject(error)
    })
    projectDataCacheMap.set(pageType, getProjectDataPromise)
  }

  return projectDataCacheMap.get(pageType)!
}
