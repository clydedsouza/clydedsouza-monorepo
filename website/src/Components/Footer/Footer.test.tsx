import {
  render,
  screen,
  waitForElementToBeRemoved,
} from '@testing-library/react'
import * as apiCacheModule from '../../Api/Cache'
import Footer from './Footer'

jest.mock('../../Api/Cache')

const ctaDataMockResponse = [
  {
    id: 'ai-tell-me-a-story',
    contents: '',
    title: 'Purchase my book AI, Tell Me a Story from these platforms.',
    hasWebsite: true,
    website: 'https://aitellmeastory.clydedsouza.net',
    excerpt: '',
  },
]

describe('Footer', () => {
  beforeEach(() => {
    jest
      .spyOn(apiCacheModule, 'getCachedProjectData')
      .mockImplementationOnce(() =>
        Promise.resolve({
          app: {},
          data: [ctaDataMockResponse[0]],
        })
      )
  })

  it('should render footer', async () => {
    render(<Footer />)
    await waitForElementToBeRemoved(() =>
      screen.queryByText('Contents are loading...')
    )
    expect(document.body).toMatchSnapshot()
  })
})
