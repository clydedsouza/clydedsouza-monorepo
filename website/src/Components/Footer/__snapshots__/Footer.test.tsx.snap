// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer should render footer 1`] = `
<body>
  <div>
    <footer>
      <div
        class="cta"
      >
        <a
          href="https://aitellmeastory.clydedsouza.net"
          rel="noreferrer"
          target="_blank"
        >
          Purchase my book AI, Tell Me a Story from these platforms.
        </a>
      </div>
      <p>
        Website crafted by Clyde DSouza
      </p>
    </footer>
  </div>
</body>
`;
